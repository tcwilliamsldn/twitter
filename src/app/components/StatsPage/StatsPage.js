import React from 'react';
import { Query } from 'react-apollo';
import RetweetGraph from './RetweetGraph/RetweetGraph';
import Loading from '../Loading/Loading';
import Error from '../Error/Error';
import getTweets from '../../queries/getTweets';
import styles from './StatsPage.css';

export const StatsPage = ({ loading, error, data }) => {
    if (loading) return <Loading />;
    if (error) return <Error message={error.message} />;

    const { tweets } = data.twitter.user;
    return (
        <div className={styles.statspage}>
            <RetweetGraph tweets={tweets} />
        </div>
    );
};

const StatsPageWithData = ({ match }) => (
    <Query query={getTweets(match.params.profileName)}>{StatsPage}</Query>
);

export default StatsPageWithData;
