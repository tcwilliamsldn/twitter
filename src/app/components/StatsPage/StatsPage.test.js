import { StatsPage } from './StatsPage';
import React from 'react';
import { shallow } from 'enzyme';

describe('StatsPage', function() {
    it('renders the loading state when loading', function() {
        const component = shallow(<StatsPage loading />);
        expect(component).toMatchSnapshot();
    });

    it('renders the error state on error', function() {
        const component = shallow(<StatsPage error={{ message: 'Whoops' }} />);
        expect(component).toMatchSnapshot();
    });

    it('renders a user Profile and Feed on success', function() {
        const data = { twitter: { user: { tweets: [] } } };

        const component = shallow(<StatsPage data={data} />);
        expect(component).toMatchSnapshot();
    });
});
