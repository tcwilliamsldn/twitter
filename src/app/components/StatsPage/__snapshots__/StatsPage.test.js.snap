// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsPage renders a user Profile and Feed on success 1`] = `
<div
  className="statspage"
>
  <RetweetGraph
    tweets={Array []}
  />
</div>
`;

exports[`StatsPage renders the error state on error 1`] = `
<Error
  message="Whoops"
/>
`;

exports[`StatsPage renders the loading state when loading 1`] = `<Loading />`;
